src/formatting/expr.rs

```diff
@@ -1895,16 +1895,22 @@ fn rewrite_assignment(
     op: Option<&ast::BinOp>,
     shape: Shape,
 ) -> Option<String> {
-    let operator_str = match op {
-        Some(op) => context.snippet(op.span),
-        None => "=",
+    let span_hi = rhs.span.lo();
+    let (operator_str, span_lo) = match op {
+        Some(op) => (context.snippet(op.span), op.span.hi()),
+        None => {
+            let lo = lhs.span.hi();
+            let offset = context.snippet(mk_sp(lo, span_hi)).find_uncommented("=")? + 1;
+            ("=", lo + BytePos(offset as u32))
+        }
```

Comment:
I could be remembering incorrectly, but I believe the AST nodes for assignment expressions actually contain the span for the operator or binop. Could you double check that (presumably back at the call sites), and if it's there then update our destructuring code to grab it?

Would much rather plug and play from the AST than recalculating spans when possible

Reply:
Yes, there is such span.  Changed the code to use it.